#Custom Search control used on DataGrid.

ipstushard opened the pull request:
#Custom Search control used on DataGrid.

ipstushard committed:
#Custom Search control used on DataGrid.

ipstushard committed:
**Apply Filter** functionality with multiple as well as single on datagrid

ShubhamG2908 commented on UtilitiesProject/Helpers.cs:
```diff
@@ -1,10 +1,199 @@
+        /// <param name="oprationKeyWord"></param>
+        /// <returns></returns>
+        /// <exception cref="NotSupportedException"></exception>
+        internal static IQueryable<T> SingleLogic<T>(IQueryable<T> query, IList<object> filterList, int propertyNamePosition, int operationPosition, int filterPosition, bool oprationKeyWord = false)
```
Rename with ApplyFilter

ipstushard replied:
Actually that fucntion is calling **ApplyFilter**.

ShubhamG2908 replied:
try "PrepareFilterQuery"

ShubhamG2908 commented on UtilitiesProject/Helpers.cs:
```diff
@@ -1,10 +1,199 @@
+                    if (filterList[propertyNamePosition].GetType().Name == typeof(string).Name)
+                    {
+                        //single filter
+                        query = SingleLogic(query, filterList, propertyNamePosition, operationPosition, filterPosition).AsQueryable();
```
I think we don't need "AsQueryable" here as you are already returning IQueryable

ipstushard replied:
I have collection of list. So, When I called **ApplyFilter** that time has Iqueryable. That time I need to convert list to IQuarable.
Anyway, let me check again to remove it and test again.

ipstushard committed:
**Apply Filter**:
Changed function name **PrepareFilterQuery** instead of **SingleLogic**
Removed **AsQueryable**

ipstushard committed:
**TreeView** :  I have created new Treeview component

ipstushard committed:
sad

ipstushard committed:
Implemented Dapper with filter query

ipstushard committed:
Merge pull request #7 from ShubhamG2908/work-with-dapper

Implemented Dapper with filter query
